core/vm/privacy/bulletproof.go

```diff
@@ -938,6 +950,10 @@ func (mrp *MultiRangeProof) Deserialize(proof []byte) error {
 	mrp.IPP.Deserialize(proof[offset:], numChallenges)
 	offset += len(mrp.IPP.L)*33 + len(mrp.IPP.R)*33 + len(mrp.IPP.Challenges)*32 + 2*32
 
+	// The last `offset += 32` will be the same value as the length of the proof
+	if len(proof) < offset+3*32 {
```

Comment:
I didn't get this part, how did len(mrp.IPP.L)*33 + len(mrp.IPP.R)*33 + len(mrp.IPP.Challenges)*32 + 2*32 become 3*32

Reply:
I think you commented on the wrong line. I think you are asking why https://github.com/XinFinOrg/XDPoSChain/pull/431/files#diff-29da4d7baf2b3531856c276e4149614fb7a999594a6f3676c5d4f4c692f9ebfaR937

I will take another look, because it indeed looks wrong because of `offset += len(mrp.IPP.L)*33 + len(mrp.IPP.R)*33 + len(mrp.IPP.Challenges)*32 + 2*32`

Reply:
@wanwiset25 i added one more check, hopefully it make sense.

The if statement is a safety check, it uses the pointer of `offset` with its future pointer value to make sure the proof has enough length so that those `proof[offset:]` methods can be successfully executed without exception.

Reply:
Right, I was confused.